dds/DCPS/RTPS/RtpsDiscovery.h

```diff
@@ -635,6 +635,66 @@ class OpenDDS_Rtps_Export RtpsDiscoveryConfig : public OpenDDS::DCPS::RcObject {
     return use_xtypes_ == XTYPES_COMPLETE;
   }
 
+  bool type_lookup_service_default_enum_extensibility(const char* str)
+  {
+    ACE_GUARD_RETURN(ACE_Thread_Mutex, g, lock_, false);
+
+    DDS::ExtensibilityKind ex = DDS::APPENDABLE;
+    if (!XTypes::TypeLookupService::parse_extensibility(str, ex)) {
+      return false;
+    }
+
+    switch (ex) {
+    case DDS::APPENDABLE:
+    case DDS::FINAL:
+      type_lookup_service_default_enum_extensibility_ = ex;
+      return true;
+    default:
+      if (log_level >= LogLevel::Warning) {
+        ACE_ERROR((LM_WARNING,
+                   "(%P|%t) WARNING: RtpsDiscoveryConfig::type_lookup_service_default_enum_extensibility: "
+                   "invalid extensibility: %C\n", str));
+      }
+      return false;
+    }
+  }
+
+  DDS::ExtensibilityKind type_lookup_service_default_enum_extensibility() const
+  {
+    ACE_GUARD_RETURN(ACE_Thread_Mutex, g, lock_, DDS::APPENDABLE);
+    return type_lookup_service_default_enum_extensibility_;
+  }
```

Comment:
Returning `APPENDABLE` in case of failing to acquire the lock will hide the error to the caller. Will the caller care if the call succeeds or not?

Reply:
If the lock cannot be acquired, the system is corrupt and all bets are off.  So, no, the caller will not care.